### ISSUE TITLE
nx-cloud depends on fs-extra but doesn't list it in dependencies

### ISSUE THREAD

**Issue by troywweber7** [nrwl/nx#12003]
## Current Behavior

`nx-cloud` requires `fs-extra` but doesn't list it in it's `package.json`

## Expected Behavior

When I start a fresh repo, I should be able to install or run `nx-cloud` without installing extra dependencies. Otherwise, `nx-cloud` is depending on hoisting packages like `fs-extra` from other installed packages and not truly supporting pnpm which recommends against hoisting.

## Steps to Reproduce

This is slightly related to https://github.com/nrwl/nx/issues/11456. However, you easily reproduce the error logs below by doing:

```bash
pnpm add -g @nrwl/nx-cloud
nx-cloud
```

### Failure Logs

```
nx-cloud 
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'fs-extra'
Require stack:
- /home/troy/.local/share/pnpm/global/5/.pnpm/@nrwl+nx-cloud@14.6.2/node_modules/@nrwl/nx-cloud/lib/utilities/distributed-task-execution-detection.js
- /home/troy/.local/share/pnpm/global/5/.pnpm/@nrwl+nx-cloud@14.6.2/node_modules/@nrwl/nx-cloud/lib/utilities/environment.js
- /home/troy/.local/share/pnpm/global/5/.pnpm/@nrwl+nx-cloud@14.6.2/node_modules/@nrwl/nx-cloud/lib/core/commands/clean-up-agents.js
- /home/troy/.local/share/pnpm/global/5/.pnpm/@nrwl+nx-cloud@14.6.2/node_modules/@nrwl/nx-cloud/bin/nx-cloud.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/troy/.local/share/pnpm/global/5/.pnpm/@nrwl+nx-cloud@14.6.2/node_modules/@nrwl/nx-cloud/lib/utilities/distributed-task-execution-detection.js:4:20)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/troy/.local/share/pnpm/global/5/.pnpm/@nrwl+nx-cloud@14.6.2/node_modules/@nrwl/nx-cloud/lib/utilities/distributed-task-execution-detection.js',
    '/home/troy/.local/share/pnpm/global/5/.pnpm/@nrwl+nx-cloud@14.6.2/node_modules/@nrwl/nx-cloud/lib/utilities/environment.js',
    '/home/troy/.local/share/pnpm/global/5/.pnpm/@nrwl+nx-cloud@14.6.2/node_modules/@nrwl/nx-cloud/lib/core/commands/clean-up-agents.js',
    '/home/troy/.local/share/pnpm/global/5/.pnpm/@nrwl+nx-cloud@14.6.2/node_modules/@nrwl/nx-cloud/bin/nx-cloud.js'
  ]
}

```

### Environment

pnpm -v 7.9.3
make sure you have hoisting turned off

**Comment by troywweber7** [comment]
@meeroslav I'm using the `.pnpmfile.cjs` trick mentioned in the other issue regarding the missing dotenv package. Since dotenv was resolved I suspect this will be as well.

I have accumulated a few more missing dependencies that break without hoisting. I'll submit separate issues for those soon.

The only thing that is really throwing me for a loop now is that even though none of us have hoisting enabled at my company, none of my colleagues (MacOS + Linux Manjaro) have an issue with this. I've been comparing their environment to mine as carefully as possible but I can't figure why theirs would be working while mine is catching these no-hoist dependency issues.

All I've been able to settle on is that if a package requires another in its source, it should probably list that in its package.json, perhaps as a peer dependency if not an actual dependency, rather that implicitly rely on it being installed by another package.
